crates/core/src/factories/core_fetch.rs

```diff
@@ -228,22 +236,19 @@ impl CoreFetch {
         while let Some((op_id, agent_id)) =
             fetch_request_rx.lock().await.recv().await
         {
-            let is_agent_cooling_down = {
+            let is_agent_on_back_off = {
                 let mut lock = state.lock().unwrap();
 
-                // Do nothing if op id is no longer in the set of ops to fetch.
-                if !lock.ops.contains(&(op_id.clone(), agent_id.clone())) {
+                // Do nothing if op id is no longer in the set of requests to send.
+                if !lock.requests.contains(&(op_id.clone(), agent_id.clone())) {
                     continue;
                 }
 
-                lock.cool_down_list.is_agent_cooling_down(&agent_id)
+                lock.back_off_list.is_agent_backing_off(&agent_id)
```

Comment:
Slightly confusing method name because we've put the agent on back off, they aren't doing the back off. Perhaps `is_agent_on_back_off`? Or `should_back_off_agent` or something like that?

Reply:
Yes, name changed to `is_agent_on_back_off`.